### ISSUE TITLE
Homematic - null java.io.IOException

### ISSUE THREAD

**Issue by FelixXXL** [openhab/openhab2-addons#2543]
Sorry for my bad english.
when i try to setup openhabbian on rpi2 &rpi3 the homematic binding can't find things.
in the log file i found this line(s).
t think that's the trouble:
can anyone help me:
```
2017-08-10 22:38:49.368 [INFO ] [ternal.communicator.client.RpcClient] - Interface 'WIRED' on gateway 'Homematic' not available, disabling support
2017-08-10 22:38:49.719 [INFO ] [ommunicator.AbstractHomematicGateway] - HmGatewayInfo[id=CCU,type=CCU2,firmware=2.29.18.20170731,address=OEQ0304821,wired=false,hmip=true,cuxd=true,group=true]
2017-08-10 22:39:15.241 [ERROR] [very.HomematicDeviceDiscoveryService] - null
java.io.IOException
        at org.openhab.binding.homematic.internal.communicator.CcuGateway.sendScript(CcuGateway.java:255)[182:org.openhab.binding.homematic:2.1.0]
        at org.openhab.binding.homematic.internal.communicator.CcuGateway.sendScriptByName(CcuGateway.java:225)[182:org.openhab.binding.homematic:2.1.0]
        at org.openhab.binding.homematic.internal.communicator.CcuGateway.sendScriptByName(CcuGateway.java:213)[182:org.openhab.binding.homematic:2.1.0]
        at org.openhab.binding.homematic.internal.communicator.CcuGateway.loadDeviceNames(CcuGateway.java:126)[182:org.openhab.binding.homematic:2.1.0]
        at org.openhab.binding.homematic.internal.communicator.AbstractHomematicGateway.getDeviceDescriptions(AbstractHomematicGateway.java:450)[182:org.openhab.binding.homematic:2.1.0]
        at org.openhab.binding.homematic.internal.communicator.AbstractHomematicGateway.loadAllDeviceMetadata(AbstractHomematicGateway.java:376)[182:org.openhab.binding.homematic:2.1.0]
        at org.openhab.binding.homematic.discovery.HomematicDeviceDiscoveryService$1.run(HomematicDeviceDiscoveryService.java:110)[182:org.openhab.binding.homematic:2.1.0]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)[:1.8.0_121]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)[:1.8.0_121]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)[:1.8.0_121]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)[:1.8.0_121]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)[:1.8.0_121]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)[:1.8.0_121]
        at java.lang.Thread.run(Thread.java:745)[:1.8.0_121]
Caused by: java.util.concurrent.TimeoutException
        at org.eclipse.jetty.client.util.FutureResponseListener.get(FutureResponseListener.java:109)[68:org.eclipse.jetty.client:9.2.19.v20160908]
        at org.eclipse.jetty.client.HttpRequest.send(HttpRequest.java:655)[68:org.eclipse.jetty.client:9.2.19.v20160908]
        at org.openhab.binding.homematic.internal.communicator.CcuGateway.sendScript(CcuGateway.java:245)[182:org.openhab.binding.homematic:2.1.0]
        ... 13 more
```

**Comment by rverst** [comment]
This may help [NPE when initiating connection to CCU2](https://github.com/openhab/openhab2-addons/issues/1501)

**TL;DR**
Since version *2.27.7* of the CCU2 the firewall is enabled automatically at system startup.

**Comment by bitsbeauty** [comment]
Having the same problem with openHAB 2.2.0-1 and piVCCU version: 2.31.25-23 on a Raspberry Pi 3 Model B Rev 1.2.
HomeMatic-Script API and/or HomeMatic XML-RPC API in the CCU is set to full access.

Any other suggestions?

**Comment by gerrieg** [comment]
piVCCU reports as a CCU2 (type=CCU2), therefore the binding sends some TCL Scripts but piVCCU can't handle them.
Configure the bridge with: gatewayType=noccu

**Comment by bitsbeauty** [comment]
Thanks, right, that worked! 
Would be nice if somebody could add a description on the Bindings page about these value. I was wondering about it before, but on the binding page it says nothing more that:

> gatewayType
>    Hint for the binding to identify the gateway type (auto|ccu|noccu) (default = auto).

**Comment by ptweety** [comment]
@gerrieg I wonder how piVCCU can not handle TCL scripts that a CCU2 has no problem with. Maybe @alexreinert want to look at this issue.

**Comment by alexreinert** [comment]
piVCCU can handle the TCL scripts without any problem, my personal instance of openHAB and piVCCU are working perfectlly together just with the option gatewayAdress set in the Bridge and gateway autodiscovery.
As piVCCU is using the original CCU firmware I have no idea how there should be issues with TCL scripts.

**Comment by mdicke2s** [comment]
@FelixXXL Is the issue that you described still valid? From the comments I wonder if there is anything to do here or not...

**Comment by mdicke2s** [comment]
This issue was moved to eclipse/smarthome#5783
